employee-rostering-frontend/src/ui/components/DataTable.tsx

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import {
+  Table,
+  TableHeader,
+  TableBody,
+  headerCol,
+  ICell,
+  IRow,
+  IAction
+} from '@patternfly/react-table';
+
+export interface DataTableProps<T> {
+  title: string;
+  columnTitles: string[];
+  tableData: T[];
+  rowDataToRow: (rowData : T) => IRow;
+}
+
+interface DataTableState<T> {
+  currentFilter: (_: T) => boolean;
+}
+
+export abstract class DataTable<T, Props extends DataTableProps<T>> extends React.Component<Props, DataTableState<T>> {
+  columns: ICell[];
+  actions: IAction[];
```

Comment:
Similar to one of my previous comments about state. Do not "cache" props like this there is no reason for that and it will only cause troubles. See below.